src/plugins/nest.js

```diff
@@ -0,0 +1,375 @@
+'use strict'
+
+function createWrapNestFactoryCreate (tracer, config) {
+  return function wrapCreate (create) {
+    return function createWithTrace (nestModule, serverOrOptions, options) {
+      const scope = tracer.scope()
+      const childOf = scope.active()
+      const span = tracer.startSpan('nest.factory.create', {
+        childOf,
+        tags: {
+          'component': 'nest',
+          'nest.module': nestModule.name
+        }
+      })
+
+      if (typeof this.createHttpAdapter === 'function') {
+        const httpServerOptions = this.isHttpServer(serverOrOptions)
```

Comment:
Are we incorporating this internal logic just for the `nest.server` tag?  If so I wonder if it's worth the potential issues as preferably we are as minimally invasive as possible.  Not seeing it tested either?

Reply:
@sm7x might be the better person to answer that?

Reply:
I only included it in an effort to get more information on the type of server. It is optional and can be removed if it would lead to potential problems.

Reply:
Also, earlier versions of nest would have broken without that check, but it may not pertain to the presently supported versions...

Reply:
Ideally we'd not directly execute client functions unless they were being traced or instrumentation is otherwise impossible.  If you find this necessary I'll defer to your judgement but a comment detailing it would be appreciated.